airflow/models/dagrun.py

```diff
@@ -1030,7 +1030,7 @@ def expand_mapped_literals(
         tasks_and_map_idxs = map(expand_mapped_literals, filter(task_filter, dag.task_dict.values()))
 
         tasks = itertools.chain.from_iterable(itertools.starmap(task_creator, tasks_and_map_idxs))
-        if missing_indexes:
+        if missing_indexes and any(len(v) for _, v in missing_indexes.items()):
```

Comment:
```suggestion
        if missing_indexes and any(len(v) for v in missing_indexes.values()):
```

Reply:
Though surely this change isn't needed as it's only passed in the one case from ti_scheduling_decisons when we've already performed this check.

Reply:
It's not the root cause of the issue but it calls verify_integrity unnecessarily